tests/conftest.py

```diff
@@ -184,26 +193,69 @@ def is_provider_installed(cls) -> bool:
 
 
 @pytest.fixture
-def create_config(tmp_path):
+def create_config(tmp_path: pathlib.Path):
```

Comment:
This is starting to get quite complex, to the point that it would be easy to modify and accidentally break some tests.

At this point I think I'd prefer if we create a data directory with various fake charms and use `shutil.copytree` to copy the relevant charm into `tmp_path`

Reply:
I though about that, but I don't think it will reduced those logic. The new `create_config()` is designed to work with/without parametrized fixtures. And if the value is not given, it will use the default value. Basically all related test cases will change some of the values that make copy and use cannot save much.